src/drivers/apl-ssp.c

```diff
@@ -536,45 +536,12 @@ static inline int ssp_set_config(struct dai *dai,
 		break;
 	case SOF_DAI_FMT_DSP_A:
 
-		start_delay = 0;
-
-		sscr0 |= SSCR0_MOD | SSCR0_FRDC(config->ssp.tdm_slots);
-
-		/* set asserted frame length */
-		frame_len = 1; /* default */
-
-		if (cfs && ssp->params.frame_pulse_width > 0 &&
-			ssp->params.frame_pulse_width <=
-			SOF_DAI_INTEL_SSP_FRAME_PULSE_WIDTH_MAX) {
-			frame_len = ssp->params.frame_pulse_width;
-		}
-
-		/* frame_pulse_width must less or equal 38 */
-		if (ssp->params.frame_pulse_width >
-			SOF_DAI_INTEL_SSP_FRAME_PULSE_WIDTH_MAX) {
-			trace_ssp_error("efa");
-			ret = -EINVAL;
-			goto out;
-		}
-
-		/*
-		 * handle frame polarity, DSP_A default is rising/active high,
-		 * non-inverted(inverted_frame=0) -- active high(SFRMP=1),
-		 * inverted(inverted_frame=1) -- falling/active low(SFRMP=0),
-		 * so, we should set SFRMP to !inverted_frame.
-		 */
-		sspsp |= SSPSP_SFRMP(!inverted_frame);
+		start_delay = 1;
 		sspsp |= SSPSP_FSRT;
 
-		active_tx_slots = hweight_32(config->ssp.tx_slots);
-		active_rx_slots = hweight_32(config->ssp.rx_slots);
-
-		sspsp2 |= (frame_end_padding & SSPSP2_FEP_MASK);
-
-		break;
```

Comment:
minor nit-pick: you are supposed to add an explicit /* fallthrough */ comment to show that you did not add a break intentionally. It's one of the things we'll enforce when we have GCC7 (also required by most FuSa/CERT rules).